File: backend/Domain/Entities/PropertyVideo.cs
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Common;

namespace Domain.Entities
{
    public class PropertyVideo : BaseEntity
    {
        [ForeignKey("Property")]
        public Guid PropertyId { get; set; }
        public required string Url { get; set; }
        
        public Property Property { get; set; } = null!;
    }
}

File: mobile/lib/src/features/dashboard/domain/entities/property_video.dart
import 'package:equatable/equatable.dart';

class PropertyVideo extends Equatable {
  final String id;
  final String url;

  const PropertyVideo({
    required this.id,
    required this.url,
  });

  @override
  List<Object?> get props => [id, url];
}
